packages/nocodb/src/db/util/durationUtils.ts

```diff
@@ -0,0 +1,128 @@
+const durationOptions = [
+  {
+    type: 0,
+    title: 'h:mm',
+    example: '(e.g. 1:23)',
+    regex: /(\d+)(?::(\d+))?/,
+  },
+  {
+    type: 1,
+    title: 'h:mm:ss',
+    example: '(e.g. 3:45, 1:23:40)',
+    regex: /(\d+)?(?::(\d+))?(?::(\d+))?/,
+  },
+  {
+    type: 2,
+    title: 'h:mm:ss.s',
+    example: '(e.g. 3:34.6, 1:23:40.0)',
+    regex: /(\d+)?(?::(\d+))?(?::(\d+))?(?:.(\d{0,4})?)?/,
+  },
+  {
+    type: 3,
+    title: 'h:mm:ss.ss',
+    example: '(e.g. 3.45.67, 1:23:40.00)',
+    regex: /(\d+)?(?::(\d+))?(?::(\d+))?(?:.(\d{0,4})?)?/,
+  },
+  {
+    type: 4,
+    title: 'h:mm:ss.sss',
+    example: '(e.g. 3.45.678, 1:23:40.000)',
+    regex: /(\d+)?(?::(\d+))?(?::(\d+))?(?:.(\d{0,4})?)?/,
+  },
+];
+
+const extractHMS = (groups: RegExpMatchArray, type: number) => {
+  if (groups[0] && groups[1] && !groups[2] && !groups[3] && !groups[4]) {
+    const val = Number.parseInt(groups[1], 10);
+    if (groups.input.slice(-1) === ':')
+      // e.g. 30:
+      return {
+        h: Number.parseFloat(groups[1]),
+        mm: 0,
+        ss: 0,
+      };
+
+    if (type === 0) {
+      // consider it as minutes
+      // e.g. 360 -> 06:00
+      const h = Math.floor(val / 60);
+      return {
+        h,
+        mm: Math.floor(val - (h * 3600) / 60),
+        ss: 0,
+      };
+    }
+    // consider it as seconds
+    // e.g. 3600 -> 01:00:00
+    return {
+      h: Math.floor(Number.parseFloat(groups[1]) / 3600),
+      mm: Math.floor(Number.parseFloat(groups[1]) / 60) % 60,
+      ss: val % 60,
+    };
+  }
+  if (type !== 0 && groups[1] && groups[2] && !groups[3])
+    // 10:10 means mm:ss instead of h:mm
+    // 10:10:10 means h:mm:ss
+    return {
+      h: 0,
+      mm: Number.parseFloat(groups[1]),
+      ss: Number.parseFloat(groups[2]),
+    };
+
+  return {
+    h: Number.parseFloat(groups[1]) || 0,
+    mm: Number.parseFloat(groups[2]) || 0,
+    ss: Number.parseFloat(groups[3]) || 0,
+  };
+};
+
+export const convertDurationToSeconds = (val: string) => {
+  const { regex, type } = durationOptions.find(({ regex }) => regex.test(val));
```

Comment:
@rohittp0 It might be a good idea to move the `durationUtils.ts` file from the frontend `/nc-gui` package to the `nocodb-sdk` and customize the convertMS2Duration function according to the use case. 
For instance, if the duration type is `undefined`, find it from `durationOptions`. This will ensure that there is only one source of truth and will reduce code duplication.

*Note*: Also after moving this utils file to `nocodb-sdk` make sure to import variable/function (`durationOptions`, `convertMS2Duration`, `convertDurationToSeconds`) in respective file of `nc-gui`

Reply:
I found no existing usage for the Date Utils in the frontend. Is there any usage or should I remove it from frontend

<img width="750" alt="image" src="https://github.com/nocodb/nocodb/assets/45072928/7038cd8e-5d59-41d4-a35f-c820602e3ba5">
<img width="750" alt="image" src="https://github.com/nocodb/nocodb/assets/45072928/8fe00a9f-3de8-4a92-9df5-c2341342cc98">

Reply:
It's already used in frontend directly without import as this is a part of global import (try searching globally).

![Screenshot 2024-05-29 at 1 58 00 AM](https://github.com/nocodb/nocodb/assets/101566080/7716070c-78f5-49cf-85c6-1aa94bf19a2e)
![Screenshot 2024-05-29 at 1 57 43 AM](https://github.com/nocodb/nocodb/assets/101566080/0c033fcb-961b-468c-a3aa-1a0a5cf17863)
![Screenshot 2024-05-29 at 1 57 30 AM](https://github.com/nocodb/nocodb/assets/101566080/734aa3e6-1828-4394-809d-c6522899f128)

Reply:
Oh sure will refactor those manually